### PR TITLE
GF-27 - Upcoming Events

### PR DESCRIPTION
**Pull request by MIKALAKE**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,9 +7,12 @@
     "@testing-library/react": "13.3.0",
     "@testing-library/user-event": "13.5.0",
     "axios": "0.27.2",
+    "country-flag-icons": "^1.5.5",
+    "moment": "^2.29.4",
```
Comment (iamdavidpetric) [inline package.json]: remove ^

File: package.json [hunk 2]
```diff
@@ -7,9 +7,12 @@
...
     "react": "18.2.0",
     "react-dom": "18.2.0",
     "react-icons": "4.4.0",
+    "react-moment": "^1.1.2",
```
Comment (iamdavidpetric) [inline package.json]: remove ^

File: src/show/components/UpcomingEvent/index.jsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+const UpcomingEvent = ({ event }) => {
+  return (
+    <div className='flex bg-charade h-44'>
+      <div className='flex ml-5 mt-3 flex-col text-white text-3xl font-bold antialiased'>
+        Upcoming Event
+        <div className='flex ml-5 mt-2 w-full h-full '>
+          <img
+            className='flex w-44 h-28'
+            src={event?.cover_url}
+            alt='circuit'></img>
```
Comment (iamdavidpetric) [inline src/show/components/UpcomingEvent/index.jsx]: one liner pls

File: src/show/pages/public/Home/index.jsx [hunk 4]
```diff
@@ -1,12 +1,22 @@
...
 
   useEffect(() => {
+    axios
+      .get('http://localhost:3000/v1/events/1')
+      .then(res => setEvent(res.data))
+      .catch(err => err);
+
     axios
       .get('http://localhost:3000/v1/constructors/')
       .then(res => setConstructors(res.data))
```
Comment (iamdavidpetric) [inline src/show/pages/public/Home/index.jsx]: do both in one useEffect

**Review by iamdavidpetric — CHANGES_REQUESTED** [review]

File: src/show/components/UpcomingEvent/index.jsx (hunk 3 above)
Comment (MIKALAKE) [reply src/show/components/UpcomingEvent/index.jsx]: u can't thx

File: src/show/pages/public/Home/index.jsx [hunk 5]
```diff
@@ -1,12 +1,22 @@
...
-import { ConstructorCard, DriverCard } from '../../../components';
+import {
+  ConstructorCard,
+  DriverCard,
+  UpcomingEvent
+} from '../../../components';
 
 const Home = () => {
   const [constructors, setConstructors] = useState([]);
+  const [event, setEvent] = useState([]);
```
Comment (iamalextodoran) [inline src/show/pages/public/Home/index.jsx]: One single event, so the initial state cannot be an empty array.

File: src/show/pages/public/Home/index.jsx [hunk 6]
```diff
@@ -1,12 +1,22 @@
...
 
   useEffect(() => {
+    axios
+      .get('http://localhost:3000/v1/events/1')
```
Comment (iamalextodoran) [inline src/show/pages/public/Home/index.jsx]:
```suggestion
      .get('http://localhost:3000/v1/events/does-not-matter')
```
So you know this does not matter, but it has to have someting after `/`. We are going to change this later on with the proper way.

File: src/show/pages/public/Home/index.jsx [hunk 7]
```diff
@@ -15,28 +25,7 @@ const Home = () => {
...
-                </div>
-              </div>
-              <div className='flex ml-10 text-white text-lg font-bold antialiased'>
-                Spain Weekend Biggest Grand Prix
-              </div>
-            </div>
-          </div>
-        </div>
-      </div>
+      <UpcomingEvent key={event.id} event={event} />
```
Comment (iamalextodoran) [inline src/show/pages/public/Home/index.jsx]: You do not need a key attribute if you do not map through multiple items.

**Review by iamalextodoran — CHANGES_REQUESTED** [review]

**Comment by iamalextodoran** [comment]
No matter if you agree or not with any comments, just leave a response for each comment you get.

File: src/show/pages/public/Home/index.jsx (hunk 5 above)
Comment (MIKALAKE) [reply src/show/pages/public/Home/index.jsx]: okay i agree

File: src/show/pages/public/Home/index.jsx (hunk 6 above)
Comment (MIKALAKE) [reply src/show/pages/public/Home/index.jsx]: So here for now I put just `does-not-matter` ?

File: src/show/pages/public/Home/index.jsx (hunk 7 above)
Comment (MIKALAKE) [reply src/show/pages/public/Home/index.jsx]: Okay, thanks

File: src/show/pages/public/Home/index.jsx (hunk 6 above)
Comment (iamalextodoran) [reply src/show/pages/public/Home/index.jsx]: Yep

File: src/show/pages/public/Home/index.jsx (hunk 5 above)
Comment (iamalextodoran) [reply src/show/pages/public/Home/index.jsx]: Still has to have an initial state. What would be suitable for this?

File: package.json [hunk 8]
```diff
@@ -7,9 +7,12 @@
     "@testing-library/react": "13.3.0",
     "@testing-library/user-event": "13.5.0",
     "axios": "0.27.2",
+    "country-flag-icons": "1.5.5",
```
Comment (iamalextodoran) [inline package.json]: You installed this library, but I don't see you using it anywhere. Remove it and run `npm install` again.

Comment (MIKALAKE) [reply package.json]: Okay, i will remove it

File: package-lock.json [hunk 9]
```diff
@@ -12,9 +12,12 @@
         "@testing-library/react": "13.3.0",
         "@testing-library/user-event": "13.5.0",
         "axios": "0.27.2",
+        "country-flag-icons": "^1.5.5",
```
Comment (iamalextodoran) [inline package-lock.json:15]: You didn't run npm install.

File: src/show/pages/public/Home/index.jsx [hunk 10]
```diff
@@ -1,12 +1,22 @@
...
-import { ConstructorCard, DriverCard } from '../../../components';
+import {
+  ConstructorCard,
+  DriverCard,
+  UpcomingEvent
+} from '../../../components';
 
 const Home = () => {
   const [constructors, setConstructors] = useState([]);
+  const [event, setEvent] = useState();
```
Comment (iamalextodoran) [inline src/show/pages/public/Home/index.jsx]: Still no initial state.

**Review by iamalextodoran — APPROVED** [review]

**Pull request merged**